misc/python/materialize/postgres_consistency/execution/pg_dialect_adjuster.py

```diff
@@ -0,0 +1,32 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+import re
+
+from materialize.output_consistency.execution.sql_dialect_adjuster import (
+    SqlDialectAdjuster,
+)
+
+
+class PgSqlDialectAdjuster(SqlDialectAdjuster):
+    def adjust_type(self, type_name: str) -> str:
+        if type_name == "DOUBLE":
+            return "DOUBLE PRECISION"
+
+        return type_name
+
+    def adjust_value(self, string_value: str, type_name: str) -> str:
+        if type_name == "INTERVAL":
+            # cut milliseconds away
+            string_value = re.sub("\\.\\d+'$", "'", string_value)
+
+        if type_name.startswith("INT") and string_value.startswith("-"):
+            # wrap negative numbers in parentheses
+            string_value = f"({string_value})"
```

Comment:
Why did you need to add this exception? If we are parsing negative numbers differently from Pg that would be a Mz product bug.

Reply:
Yes, I will file an issue.

The evaluation order is different between Postgres and mz. This results in an out-of-range error in Postgres but does not in mz.

Postgres evaluates `SELECT -2147483648::INT` as `SELECT -(2147483648::INT)` while mz does `SELECT (-2147483648)::INT`. `(2147483648::INT)` is out of integer range.

Reply:
✅ Filed https://github.com/MaterializeInc/materialize/issues/21993.